test/integration/api/compact_index_test.rb

```diff
@@ -2,107 +2,153 @@
+2.0.0 gemA1:= 1.0.0|checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,rubygems:>= 2.6.3
+1.2.0 |checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,ruby:>= 2.0.0,rubygems:>1.9
+2.1.0 gemA1:= 1.0.0,gemA2:= 1.0.0|checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,ruby:>= 2.0.0,rubygems:>=2.0
+eos
 
-    rubygem = create(:rubygem, name: 'AAA')
-    create(:version, rubygem: rubygem, number: '1.0.0')
+    get info_path(gem_name: 'gemA')
 
-    get versions_path
     assert_response :success
-    assert_match(/ZZZ 1.0.0 \nAAA/, @response.body)
+    assert_equal expected, @response.body
+    assert_equal etag(expected), @response.headers['ETag']
+    assert_equal expected, CompactIndex.info(Rails.cache.read("info/gemA"))
   end
 
-  test "/versions partial response" do
-    get versions_path, nil, range: "bytes=200-"
+  test "/info partial response" do
+    expected = <<-eos
+---
+1.0.0 |checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,rubygems:>= 2.6.3
+2.0.0 gemA1:= 1.0.0|checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,rubygems:>= 2.6.3
+1.2.0 |checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,ruby:>= 2.0.0,rubygems:>1.9
+2.1.0 gemA1:= 1.0.0,gemA2:= 1.0.0|checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,ruby:>= 2.0.0,rubygems:>=2.0
+eos
+
+    get info_path(gem_name: 'gemA'), nil, range: "bytes=159-"
+
     assert_response 206
-    assert_equal "0.0 \ngemA 1.2.0 \ngemA 2.0.0 \ngemA 2.1.0 \ngemB 1.0.0 \n", @response.body
+    assert_equal expected[159..-1], @response.body
   end
 
-  test "/info with existing gem" do
-    get info_path(gem_name: @rubygem.name)
+  test "/info with new gem" do
+    rubygem = create(:rubygem, name: 'gemC')
+    version = create(:version, rubygem: rubygem, number: '1.0.0', info_checksum: '65ea0d')
+    create(:dependency, :development, version: version, rubygem: @rubygem2)
+    expected = <<-END
+---
+1.0.0 |checksum:b5d4045c3f466fa91fe2cc6abe79232a1a57cdf104f7a26e716e0a1e2789df78,ruby:>= 2.0.0,rubygems:>= 2.6.3
+END
+
+    get info_path(gem_name: 'gemC')
+
     assert_response :success
-    assert_equal "---\n" \
-      "1.0.0 |checksum:checksum1,rubygems:>= 2.6.3\n" \
-      "1.2.0 |checksum:checksum1,ruby:>= 2.0.0,rubygems:>1.9\n" \
-      "2.0.0 gemA1:= 1.0.0|checksum:checksum2,rubygems:>= 2.6.3\n" \
-      "2.1.0 gemA1:= 1.0.0,gemA2:= 1.0.0|checksum:checksum2,ruby:>= 2.0.0,rubygems:>=2.0\n",
-      @response.body
+    assert_equal(expected, @response.body)
+    assert_equal etag(expected), @response.headers['ETag']
   end
 
-  test "/info with unexisting gem" do
+  test "/info with nonexistent gem" do
     get info_path(gem_name: 'donotexist')
     assert_response :not_found
-  end
-
-  test "/info partial response" do
-    get info_path(gem_name: @rubygem.name), nil, range: "bytes=159-"
-    assert_response 206
```

Comment:
@fotanus It is not removed, it is moved up [here](https://github.com/rubygems/rubygems.org/pull/1327/files#diff-a708a123bd49a50ac54c92bdcdc6028dR129).